releng/tools/rmake

```diff
@@ -7,24 +7,38 @@ import multiprocessing
 from common import *
 
 usage="""Usage:
-    rmake --all                Compile all configured suites
-    rmake --list               Show available compilation targets
-    rmake --doc                Build & serve documentation (rebuilds on every change)
-    rmake --help               Print this help
-    rmake <targetName>...      Compile one or multiple targets
-    rmake <pathPrefix>...      Compile one or multiple targets in path
-    rmake <suitename>...       Compile given suite
-    rmake clean                Clean all suite builds
-    rmake clean-<suitename>... Clean given suite
+    rmake --all                  Compile all configured suites
+
+    rmake --list                 Show available compilation targets
+    rmake --doc                  Build & serve documentation (rebuilds on every change)
+    rmake --help                 Print this help
+
+    rmake <targetName>...        Compile one or multiple targets
+    rmake <pathPrefix>...        Compile one or multiple targets in path
+    rmake <suitename>...         Compile given suite
+
+    rmake --test --all           Compile and run all configured tests
+    rmake --test <targetName>... Compile and run test for target
+    rmake --test <pathName>...   Compile and run tests from given path prefix
+    rmake --test <suitename>...  Compile and run tests from given suite
+
+    rmake clean                  Clean all suite builds
+    rmake clean-<suitename>...   Clean given suite
 """
 
-def compileTarget(target):
-    print(f"Compiling target: {target}")
-    try:
-        subprocess.check_call(["make", f"-j{multiprocessing.cpu_count()}", target.target],
-            cwd=os.path.join(buildDir, target.suite))
-    except subprocess.SubprocessError as e:
-        sys.exit(f"Build of target {target} failed. See output above\n")
+def compileTargets(targets):
+    """
+    Compile given targets. Be as parallel as possible
+    """
+    for s in configuredSuites():
+        sTargets = [x.target for x in targets if x.suite == s]
+        print(f"Compiling targets from suite {s}: {', '.join(sTargets)}")
+        try:
+            threads = int(1.5 * multiprocessing.cpu_count())
```

Comment:
I don't understand this line, but it can be ok

Reply:
Which one? None is highlighted.

Reply:
the 37 (if you canhighlight one, I don't know how)

Reply:
Ah. That is correct. The 1.5 multiplier is to mask I/O starvation by running other compilation jobs. But I guess it could be smaller - e.g., 1.1 or a constant (1 or 2)